pinot-query-runtime/src/main/java/org/apache/pinot/query/runtime/operator/LeafStageTransferableBlockOperator.java

```diff
@@ -84,11 +89,18 @@ protected TransferableBlock getNextBlock() {
     } else {
       if (_currentIndex < _baseResultBlock.size()) {
         InstanceResponseBlock responseBlock = _baseResultBlock.get(_currentIndex++);
-        BaseResultsBlock resultsBlock = responseBlock.getResultsBlock();
-        if (resultsBlock != null) {
-          List<Object[]> rows =
-              toList(resultsBlock.getRows(responseBlock.getQueryContext()), responseBlock.getDataSchema());
-          return new TransferableBlock(rows, responseBlock.getDataSchema(), DataBlock.Type.ROW);
+        if (responseBlock.getResultsBlock() != null) {
+          DataSchema dataSchema = responseBlock.getDataSchema();
+          boolean requiresCleanup = responseBlock.getResultsBlock() instanceof SelectionResultsBlock
```

Comment:
nit can we break this out into its own method? will be a bit easier if we need to set debugger breakpoints and what not

Reply:
yes agreed with you, this is refactor out to one per baseresulttype